rotkehlchen/chain/evm/tokens.py

```diff
@@ -201,6 +211,45 @@ def _query_chunks(
             total_token_balances = combine_dicts(total_token_balances, new_token_balances)
         return total_token_balances
 
+    def _compute_detected_tokens_info(self, addresses: list[ChecksumEvmAddress]) -> DetectedTokensType:  # noqa: E501
+        addresses_info: DetectedTokensType = defaultdict(lambda: (None, None))
+        # Need a mapping of proxy -> owner address
+        reversed_proxies_mappig = {v: k for k, v in self.proxies_mapping.items()}
+        with self.db.conn.read_ctx() as cursor:
+            for address in addresses:
+                # If `address` is a proxy, save token balances to its owner address
+                proxied_address = reversed_proxies_mappig.get(address, address)
```

Comment:
I believe that part of my confusion in this logic comes from this variable. It might be the proxied address or the proxy address.

Doesn't your logic need in the case where you have both address and a proxied address to save two entries in the `addresses_info` dict?

Reply:
No, this since this is the exact point of `_compute_detected_tokens_info`. To have tokens of proxies visible as tokens of the proxies' owners. This is needed for more convenient behaviour of the app